src/Uno.UI/ts/Windows/Storage/ApplicationDataContainer.ts

```diff
@@ -10,6 +10,34 @@ namespace Windows.Storage {
 			return `UnoApplicationDataContainer_${locality}_`;
 		}
 
+		// https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API/Using_the_Web_Storage_API#feature-detecting_localstorage
+		private static isLocalStorageAvailable() {
+			let storage: Storage;
+			try {
+				storage = window['localStorage'];
+				const x = "__storage_test__";
+				storage.setItem(x, x);
+				storage.removeItem(x);
+				return true;
+			} catch (e) {
+				return (
+					e instanceof DOMException &&
+					// everything except Firefox
+					(e.code === 22 ||
+						// Firefox
+						e.code === 1014 ||
+						// test name field too, because code might not be present
+						// everything except Firefox
+						e.name === "QuotaExceededError" ||
+						// Firefox
+						e.name === "NS_ERROR_DOM_QUOTA_REACHED") &&
+					// acknowledge QuotaExceededError only if there's something already stored
+					storage &&
+					storage.length !== 0
+				);
+			}
```

Comment:
We probably should be doing this only once for perf reasons, and log if it's not enabled.

Reply:
It depends... if the error is coming from a quota reached, it means it could be available in the future.

Maybe the check should be done while actually accessing the dom storage instead of preemptively set/remove an item on it.

Reply:
@carldebilly @jeromelaban Actually, I think we can indeed just cache this.

quota reached isn't a concern. It looks like the code **does** return true for this case.